### ISSUE TITLE
killing a process

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11187]
|<img alt="synapsis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| synapsis reported|
|-|-|
|Trac ID|trac#1694|
|Type|defect|
|Created|2006-05-02 20:07:56Z|

In the HOWTO documentation, "Using Processes", it is reported that to kill a process, one have to do: os.kill(self.transport.pid, signal.SIGKILL).

At least on Windows the pid attribute (and os.kill, of course) does not exists.

Moreover it seems that transport now(?) has a portable method: signalProcess
that works both on Posix and Windows.

This should be documented.

<details><summary>Searchable metadata</summary>

```
trac-id__1694 1694
type__defect defect
reporter__synapsis synapsis
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1146600476000000 1146600476000000
changetime__1158989868000000 1158989868000000
version__ 
owner__ 
cc__spiv
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set owner to edsuominen|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set owner to edsuom|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @glyph<br>@glyph set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph removed owner<br>@glyph set status to `new`|
|-|-|

The change, which is minimal, is in ```pid-1694```.

The buildbot should be done verifying that it didn't introduce any more doc warnings soon.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

Buildbot finished, and there were only 10 warnings, as before.  For a little while at least, the link will be here:

http://twistedmatrix.com/buildbot/full-2.3/builds/1767/step-process-docs/1

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @jml|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml set owner to @glyph|
|-|-|

Looks good, please merge

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph set status to `closed`|
|-|-|

(In [18261]) Remove reference to os.kill in process documentation.

Author: glyph

Reviewer: jml

Fixes [#11187](https://github.com/twisted/twisted/issues/11187)

The documentation incorrectly advised users to call 'os.kill' on a
ProcessTransport's 'pid' attribute if they wanted to kill it.  This is
unportable and unreliable.  This changes the documentation to refer to the
signalProcess Twisted API instead, which works (insofar as signals can be said
to work at all) on Windows.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
